src/SourceBuild/content/eng/finish-source-only.proj

```diff
@@ -30,16 +30,46 @@
                           OutputPath="$(ArtifactsLogDir)" />
   </Target>
 
+  <UsingTask TaskName="IndexItemGroup" TaskFactory="RoslynCodeTaskFactory" AssemblyFile="$(MSBuildToolsPath)\Microsoft.Build.Tasks.Core.dll" >
+    <ParameterGroup>
+      <Items Required="true" ParameterType="Microsoft.Build.Framework.ITaskItem[]"/>
+      <Index Required="true" ParameterType="System.Int32"/>
+      <Item Output="true" ParameterType="Microsoft.Build.Framework.ITaskItem"/>
+    </ParameterGroup>
+    <Task>
+      <Code Type="Fragment" Language="cs">
+        <![CDATA[Item = Items[ Index ];]]>
+      </Code>
+    </Task>
+  </UsingTask>
```

Comment:
I think this would be our first inline C# msbuild task. Could we just a regex pattern below which is already natively supported as an msbuild function?

Reply:
Actually, I see that you just want to read a specific line. I think that should already be supported in msbuild without a task. cc @rainersigwald

Reply:
Also see https://stackoverflow.com/questions/29403937/is-there-an-indexer-available-in-msbuild-itemgroups (the Edit part in the initial comment)

Reply:
We do use inline msbuild tasks in other places, in VMR. But, I like the idea of avoiding the task if possible - will check the other suggested solution.

Reply:
Sounds good. Are you sure that the VMR uses inline tasks? I can't find any hits: https://github.com/search?q=repo%3Adotnet%2Fsdk%20RoslynCodeTaskFactory&type=code

Reply:
Fixed with https://github.com/dotnet/sdk/pull/44362/commits/2c1e9e20d17b5f667e4ac65a6599841129b9d158

Reply:
Example inline tasks usage in VMR: https://github.com/dotnet/dotnet/blob/5ab27b3f962248d6a010146dac4ab6545d4c8f05/src/arcade/src/Microsoft.DotNet.Arcade.Sdk/tools/Workarounds.targets#L191-L194

Reply:
OK I thought you meant our infra that we own, the VMR orchestrator, That one is a workaround thing that will be removed again in ~ a month.